Distributed code should not contain webpack specific logic

There is a require statement in the client https://github.com/rethinkdb/horizon/blob/next/client/src/util/fetch.js#L6 which is webpack specific and therefore when I attempt to build using typescript I come up against the following error:

```
Error: Cannot find module 'imports?this=>global!exports?global.fetch!isomorphic-fetch' from '/Users/home/demo/node_modules/@horizon/client/lib/util'
```

--- a/client/package.json
+++ b/client/package.json
@@ -16,9 +16,8 @@
   "dependencies": {
     "babel-runtime": "^6.6.1",
     "core-js": "^2.1.0",
-    "exports-loader": "^0.6.3",
-    "imports-loader": "^0.6.5",
-    "isomorphic-fetch": "^2.2.1",
+    "whatwg-fetch": "^1.0.0",
+    "es6-promise": "^3.2.1",
     "rxjs": "5.0.0-beta.7",
     "snake-case": "^1.1.2",
     "ws": "^1.1.0"
@@ -40,6 +39,8 @@
     "copy-webpack-plugin": "^1.1.1",
     "cross-env": "^1.0.7",
     "eslint": "2.2.x",
+    "exports-loader": "^0.6.3",
+    "imports-loader": "^0.6.5",
     "istanbul": "^0.4.2",
     "lodash": "^4.6.1",
     "mocha": "2.3.4",

--- a/client/package.json
+++ b/client/package.json
@@ -39,8 +39,6 @@
     "copy-webpack-plugin": "^1.1.1",
     "cross-env": "^1.0.7",
     "eslint": "2.2.x",
-    "exports-loader": "^0.6.3",
-    "imports-loader": "^0.6.5",
     "istanbul": "^0.4.2",
     "lodash": "^4.6.1",
     "mocha": "2.3.4",

--- a/client/package.json
+++ b/client/package.json
@@ -39,6 +39,8 @@
     "copy-webpack-plugin": "^1.1.1",
     "cross-env": "^1.0.7",
     "eslint": "2.2.x",
+    "exports-loader": "^0.6.3",
+    "imports-loader": "^0.6.5",
     "istanbul": "^0.4.2",
     "lodash": "^4.6.1",
     "mocha": "2.3.4",

--- a/client/src/util/fetch.js
+++ b/client/src/util/fetch.js
@@ -2,9 +2,6 @@ import { Observable } from 'rxjs/Observable'
 import 'rxjs/add/observable/fromPromise'
 import 'rxjs/add/operator/mergeMap'
 
-global.self = global
-require('imports?this=>global!exports?global.fetch!isomorphic-fetch')
-
 export default function fetchJSON(url) {
   return Observable.fromPromise(fetch(url))
     .mergeMap(response => {

--- a/client/webpack.horizon.config.js
+++ b/client/webpack.horizon.config.js
@@ -6,6 +6,7 @@ const DefinePlugin = require('webpack/lib/DefinePlugin')
 const OccurrenceOrderPlugin = require(
   'webpack/lib/optimize/OccurrenceOrderPlugin')
 const UglifyJsPlugin = require('webpack/lib/optimize/UglifyJsPlugin')
+const ProvidePlugin = require('webpack/lib/ProvidePlugin');
 
 module.exports = function(buildTarget) {
   const FILENAME = buildTarget.FILENAME
@@ -83,6 +84,10 @@ module.exports = function(buildTarget) {
       // and thus strip helpful warnings from production build:
       new DefinePlugin({
         'process.env.NODE_ENV': (DEV_BUILD ? 'development' : 'production'),
+      }),
+      new ProvidePlugin({
+        'Promise': 'es6-promise',
+        'fetch': 'imports?this=>global!exports?global.fetch!whatwg-fetch'
       }),
     ].concat(DEV_BUILD ? [] : [
       new DedupePlugin(),

